### PR TITLE
Added endpoint for clearing all completed task, added auto-generated created date and updated date

### PR DESCRIPTION
**Pull request by nhatnam23012000**
Now the created date and updated date will be automatically generated on POST and PUT methods, I also add a new endpoint for clearing all the completed according the example website

### CODE REVIEW

File: controllers/controller.js [hunk 1]
```diff
@@ -1,30 +1,55 @@
...
+  const tasks = await TaskModel.find()
+  res.json(tasks)
 }
 
 // add new task to database
 module.exports.addTask = async (req, res) => {
-  const tasks = await TaskModel.create(req.body)
-  res.json(tasks);
+  const title = req.body.title
+  const isCompleted = req.body.isCompleted
```
Comment (honglax) [inline controllers/controller.js]: `const {title, isCompleted} = req.body`

File: controllers/controller.js [hunk 2]
```diff
@@ -1,30 +1,55 @@
...
+  const createdAt = new Date().toLocaleString()
+  const updatedAt = ''
```
Comment (honglax) [inline controllers/controller.js]: As I told you, `updatedAt` must be generated when creating a new task. 

`const updatedAt = new Date().toLocaleString()`

File: controllers/controller.js [hunk 3]
```diff
@@ -1,30 +1,55 @@
...
+  const tasksTobeAdded = new TaskModel({
+    title,
+    isCompleted,
+    createdAt,
+    updatedAt,
+  })
+
+  const tasks = await tasksTobeAdded.save()
```
Comment (honglax) [inline controllers/controller.js]: Please use mongoose API create()
`const task = TaskModel.create({
  title,
  isCompleted,
  createdAt,
  updatedAt
})`

Comment (honglax) [reply controllers/controller.js]: Variable `task` must be single, not plural.

File: controllers/controller.js [hunk 4]
```diff
@@ -1,30 +1,55 @@
...
-    return res.json(tasks);
-  });
+  const dataToUpdate = req.body
+  dataToUpdate.updatedAt = new Date().toLocaleString()
+  await TaskModel.findByIdAndUpdate(
+    req.params.id,
+    dataToUpdate,
+    (err, tasks) => {
+      if (err) return res.status(500).json(err)
+      return res.json(tasks)
```
Comment (honglax) [inline controllers/controller.js]: Variable `task` must be single, not plural.

File: controllers/controller.js [hunk 5]
```diff
@@ -1,30 +1,55 @@
...
 module.exports.updateTask = async (req, res) => {
-  await TaskModel.findByIdAndUpdate(req.params.id, req.body, (err, tasks) => {
-    if (err) return res.status(500).json(err);
-    return res.json(tasks);
-  });
+  const dataToUpdate = req.body
+  dataToUpdate.updatedAt = new Date().toLocaleString()
+  await TaskModel.findByIdAndUpdate(
+    req.params.id,
+    dataToUpdate,
```
Comment (honglax) [inline controllers/controller.js]:
```
const {id} = req.body
const updatedTask = {
   ...req.body,
   updatedAt: new Date().toLocaleString()
}

await TaskModel.findByIdAndUpdate(id, updatedTask, ...)
```

File: controllers/controller.js [hunk 6]
```diff
@@ -1,30 +1,55 @@
...
+    return res.json(message)
+  })
+}
+
+// clear completed tasks
+module.exports.deleteCompleted = async (req, res) => {
+  await TaskModel.deleteMany( {isCompleted: true}, (err, tasks) => {
+    if (err) return res.status(500).json(err)
+    const message = 'Clear completed tasks'
+    return res.json(message)
```
Comment (honglax) [inline controllers/controller.js]: As I told you, we must define the API as simple as possible. I don't think it's a good idea when you split the delete endpoint. If you want to delete only the completed task, please handle it on the front-end side.

File: routers/router.js [hunk 7]
```diff
@@ -7,5 +7,6 @@ router.get('/', TaskController.findAllTasks)
 router.post('/', TaskController.addTask)
 router.put('/:id', TaskController.updateTask)
 router.delete('/:id', TaskController.deleteTask)
+router.delete('/', TaskController.deleteCompleted)
```
Comment (honglax) [inline routers/router.js]: Remove this route. As I told you above, we don't need to define another function to delete only completed tasks. Please handle it on the front-end side. Besides that, the way you define a delete method to the '/' path is very dangerous. We never do this.

**Review by honglax — CHANGES_REQUESTED** [review]

File: controllers/controller.js (hunk 1 above)
Comment (nhatnam23012000) [reply controllers/controller.js]: I fixed this mistake

File: controllers/controller.js (hunk 2 above)
Comment (nhatnam23012000) [reply controllers/controller.js]: I fixed this mistake

File: controllers/controller.js (hunk 3 above)
Comment (nhatnam23012000) [reply controllers/controller.js]: I fixed this mistake

File: controllers/controller.js (hunk 6 above)
Comment (nhatnam23012000) [reply controllers/controller.js]: I fixed this mistake

File: controllers/controller.js (hunk 4 above)
Comment (nhatnam23012000) [reply controllers/controller.js]: I fixed this mistake

File: controllers/controller.js (hunk 5 above)
Comment (nhatnam23012000) [reply controllers/controller.js]: I fixed this mistake

File: routers/router.js (hunk 7 above)
Comment (nhatnam23012000) [reply routers/router.js]: I'm sorry about this, I fixed this

File: controllers/controller.js [hunk 8]
```diff
@@ -1,30 +1,43 @@
...
+    return res.json(task)
+  })
 }
 
 // delete tasks
 module.exports.deleteTask = async (req, res) => {
   await TaskModel.findByIdAndRemove(req.params.id, (err, tasks) => {
-    if (err) return res.status(500).json(err);
+    if (err) return res.status(500).json(err)
     const message = 'Deleted task'
```
Comment (honglax) [inline controllers/controller.js]: The message should be 
```
`The task with id ${req.params.id} has been deleted successfully.`
```

Comment (nhatnam23012000) [reply controllers/controller.js]: I fixed this problem

**Review by honglax — APPROVED** [review]
LGTM

**Pull request merged**